### ISSUE TITLE
Open Collapsed Folds?

### ISSUE THREAD

**Issue by ph1lh00k** [Dook1/Bookmarks-Dook#14]
Hi @Dook1,

Firstly, thanks for providing this plugin - it's really helped me in my workflow!  =D

I was just wondering if it would be possible for the plugin to unfold to the target bookmark if it's hidden by folding?

Thanks,
Phil

**Comment by Dook1** [comment]
Explain in more detail what you mean.

**Comment by ph1lh00k** [comment]
Hi @Dook1,

Sorry for the delayed response.

I tend to Fold/Collapse different levels of my code when I have a large file to make it easier to navigate.

For example :
```
    class Class():
	    def Function():
		    code
>		    if condition:   (bookmarked here)
			    code
```

Could be collapsed to :

```
class Class(): ...
```

If I trigger the bookmark for the "if condition:" line when Class is collapsed, it will just take me to the Class line.

I was hoping it might be possible for the plugin to unfold/uncollapse the levels necessary to display the "if condition:" line...?

Thanks,
Phil

**Comment by Keyhabit** [comment]
Many thanks for providing this excellent and much-needed plugin!

Another Phil(ipp) here, who'd like to second this request to be able to navigate to bookmarks when they are inside a folded code section.

Currently when clicking on a folded-away bookmark in the Dook panel, nothing will happen or the view will be set to some area above the folded target, I think. As a workaround for getting to the bookmarked line, I have to take a mental note of the line number (I have them shown in the Dook panel), press Ctrl-g, enter the number and press Enter. In this was, the bookmarked line is shown successfully. Whatever method the plugin currently uses internally does not work in the same way apparently and can only jump to bookmarks that are visible on the top level of folded code.

Please let me know if you want me to post / send a screencast illustrating the issue.

Many thanks!

**Comment by Keyhabit** [comment]
Me again. Perhaps [this post](https://community.notepad-plus-plus.org/post/77234) might shed some light on how to fix the issue?

**Comment by Dook1** [comment]
Ok. I'll look into it as soon as I'm free.

**Comment by Keyhabit** [comment]
Great, thanks!

**Comment by Dook1** [comment]
fixed in 3.0.0
https://github.com/Dook1/Bookmarks-Dook/releases/tag/3.0
